upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -198,6 +201,11 @@ func (c *ApplyClusterCmd) Run() error {
 		return fmt.Errorf("unknown phase %q", c.Phase)
 	}
 
+	stageAssetsLifecycle = c.overrideLifecycle(PhaseStageAssets, stageAssetsLifecycle)
```

Comment:
Looks like it would be simpler to pass in the map of phases to lifecycle values into applycluster.

Then as you parse them overrides, you can simply set the overrides.

Also, we know we're going to want more sophisticated logic in future - a lifecycle oracle class.

Reply:
Need a bit more context here.

> Looks like it would be simpler to pass in the map of phases to lifecycle values into applycluster.

I am passing in:

`LifecycleOverrides map[Phase]fi.Lifecycle`

Which is a map of phases => lifecycle.  Not sure what you are recommending.